File: Ad-Hoc/1107/1107.py
# A ideia é somar todas as quedas de altura.
# Primeiro contamos a queda inicial (do topo até a primeira posição).
# Depois, para cada posição, se a altura for menor que a anterior,
# significa que o laser precisou ligar para descer, então somamos essa diferença.

class LaserSculpture:
    @staticmethod
    def count_turns_on(height, length):
        heights = [int(x) for x in input().split()]
        response = height - heights[0]  # queda inicial

        for i in range(1, length):
            if heights[i] < heights[i-1]:
                response += heights[i-1] - heights[i]

        print(response)


while True:
    height, length = [int(x) for x in input().split()]
    if height == length == 0: break
    LaserSculpture.count_turns_on(height, length)

File: Ad-Hoc/__1000__/1107/1107.cpp
#include <iostream>
#include <vector>
using namespace std;

// A ideia é somar todas as quedas de altura.
// Primeiro contamos a queda inicial (do topo até a primeira posição).
// Depois, para cada posição, se a altura for menor que a anterior,
// significa que o laser precisou ligar para descer, então somamos essa diferença.
int laser_sculture(int height, int length, vector<int>& heights) {
    int response = height - heights[0];  
    for (int i = 1; i < length; i++) {
        if (heights[i] < heights[i-1]) {
            response += heights[i-1] - heights[i];
        }
    }
    return response;
}

int main() {
    int height, length;
    while (cin >> height >> length) {
        if (height == 0 && length == 0) break;

        vector<int> heights(length);
        for (int i = 0; i < length; i++) {
            cin >> heights[i];
        }

        int out = laser_sculture(height, length, heights);
        cout << out << endl;
    }
    return 0;
}